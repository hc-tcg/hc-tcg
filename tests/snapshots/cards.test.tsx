import {describe, expect, test} from '@jest/globals'
import EffectCard from 'client/components/card/effect-card-svg'
import HermitCard from 'client/components/card/hermit-card-svg'
import ItemCard from 'client/components/card/item-card-svg'
import EthosLabCommon from 'common/cards/default/hermits/ethoslab-common'
import BuilderDoubleItem from 'common/cards/default/items/builder-rare'
import GoldenAxe from 'common/cards/default/single-use/golden-axe'
import * as ReactTestRenderer from 'react-test-renderer'

describe('Test card snapshots', () => {
	test('Hermit card SVG', () => {
		const render = ReactTestRenderer.create(
			<HermitCard card={EthosLabCommon} displayTokenCost={true} />,
		)
		expect(render.toJSON()).toMatchSnapshot()
	})
	test('Item Card SVG', () => {
		const render = ReactTestRenderer.create(
			<ItemCard card={BuilderDoubleItem} displayTokenCost={true} />,
		)
		expect(render.toJSON()).toMatchSnapshot()
	})
	test('Effect Card SVG', () => {
		const render = ReactTestRenderer.create(
			<EffectCard card={GoldenAxe} displayTokenCost={true} />,
		)
		expect(render.toJSON()).toMatchSnapshot()
	})
})
