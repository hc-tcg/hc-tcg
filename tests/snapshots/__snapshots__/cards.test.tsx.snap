// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test card snapshots Effect Card SVG 1`] = `
<svg
  className=""
  height="100%"
  viewBox="0 0 400 400"
  width="100%"
>
  <rect
    height="380"
    rx="15"
    ry="15"
    width="380"
    x="10"
    y="10"
  />
  <g>
    <image
      href="/images/star_color.svg"
      width="390"
      x="-15"
      y="65"
    />
    <image
      height="220"
      href="/images/effects/golden_axe.png"
      width="220"
      x="90"
      y="132"
    />
  </g>
  <g
    id="type"
  >
    <rect
      height="75"
      rx="15"
      ry="15"
      width="360"
      x="20"
      y="20"
    />
    <text
      dominantBaseline="hanging"
      textAnchor="middle"
      x="200"
      y="33"
    >
      EFFECT
    </text>
  </g>
  <g>
    <rect
      height="100"
      rx="50"
      ry="50"
      width="100"
      x="0"
      y="302"
    />
    <image
      height="70"
      href="/images/ranks/gold.png"
      width="70"
      x="15"
      y="315"
    />
  </g>
  <defs>
    <filter
      colorInterpolationFilters="sRGB"
      height="200%"
      id="drop-shadow"
      width="200%"
      x="-50%"
      y="-50%"
    >
      <feGaussianBlur
        id="blur"
        in="SourceAlpha"
        result="SA-o-blur"
        stdDeviation="5"
      />
      <feComponentTransfer
        in="SA-o-blur"
        result="SA-o-b-contIN"
      >
        <feFuncA
          id="contour"
          tableValues="0 1"
          type="table"
        />
      </feComponentTransfer>
      <feComposite
        in="SA-o-blur"
        in2="SA-o-b-contIN"
        operator="in"
        result="SA-o-b-cont"
      />
      <feComponentTransfer
        in="SA-o-b-cont"
        result="SA-o-b-c-sprd"
      >
        <feFuncA
          id="spread-ctrl"
          slope="200"
          type="linear"
        />
      </feComponentTransfer>
      <feColorMatrix
        id="recolor"
        in="SA-o-b-c-sprd"
        result="SA-o-b-c-s-recolor"
        type="matrix"
        values="0 0 0 0 1 0 0 0 0 1 0 0 0 0 1 0 0 0 1 0"
      />
      <feMerge>
        <feMergeNode
          in="SA-o-b-c-s-r-mix"
        />
        <feMergeNode
          in="SourceGraphic"
        />
      </feMerge>
    </filter>
  </defs>
</svg>
`;

exports[`Test card snapshots Hermit card SVG 1`] = `
<svg
  className=""
  height="100%"
  viewBox="0 0 400 400"
  width="100%"
>
  <defs>
    <clipPath
      id="myClip"
    >
      <rect
        height="178"
        width="290"
        x="55"
        y="70"
      />
    </clipPath>
  </defs>
  <rect
    className=""
    height="380"
    rx="15"
    ry="15"
    width="380"
    x="10"
    y="10"
  />
  <text
    className=""
    dominantBaseline="hanging"
    lengthAdjust="spacingAndGlyphs"
    textLength=""
    x="45"
    y="20"
  >
    Etho
  </text>
  <text
    dominantBaseline="hanging"
    textAnchor="middle"
    x="310"
    y="20"
  >
    260
  </text>
  <g
    id="hermit-image"
  >
    <rect
      fill="white"
      height="196"
      width="310"
      x="45"
      y="60"
    />
    <image
      clipPath="url(#myClip)"
      href="/images/backgrounds/ethoslab.png"
      width="290"
      x="55"
      y="70"
    />
    <image
      clipPath="url(#myClip)"
      href="/images/hermits-nobg/ethoslab.png"
      width="290"
      x="55"
      y="70"
    />
  </g>
  <g
    id="hermit-type"
  >
    <rect
      height="100"
      rx="50"
      ry="50"
      width="100"
      x="315"
      y="-5"
    />
    <image
      height="68"
      href="/images/types/type-balanced.png"
      width="68"
      x="327"
      y="12"
    />
  </g>
  <g>
    <image
      height="70"
      href="/images/ranks/iron.png"
      width="70"
      x="68"
      y="80"
    />
  </g>
  <g
    id="hermit-attacks"
  >
    <g>
      <image
        className=""
        height={28}
        href="/images/types/type-any.png"
        width={28}
        x={48}
        y="273"
      />
    </g>
    <text
      className=""
      dominantBaseline="hanging"
      textAnchor="middle"
      x="200"
      y="272"
    >
      Snack Time
    </text>
    <text
      className=""
      dominantBaseline="hanging"
      textAnchor="middle"
      x="380"
      y="270"
    >
      40
    </text>
    <rect
      fill="white"
      height="10"
      width="360"
      x="20"
      y="315"
    />
    <image
      className=""
      height={28}
      href="/images/types/type-balanced.png"
      width={28}
      x={20}
      y="343"
    />
    <image
      className=""
      height={28}
      href="/images/types/type-balanced.png"
      width={28}
      x={48}
      y="343"
    />
    <image
      className=""
      height={28}
      href="/images/types/type-balanced.png"
      width={28}
      x={76}
      y="343"
    />
    <text
      className=""
      dominantBaseline="hanging"
      textAnchor="middle"
      x="200"
      y="342"
    >
      Breach
    </text>
    <text
      className=""
      dominantBaseline="hanging"
      textAnchor="middle"
      x="380"
      y="340"
    >
      100
    </text>
  </g>
</svg>
`;

exports[`Test card snapshots Item Card SVG 1`] = `
<svg
  height="100%"
  viewBox="0 0 400 400"
  width="100%"
>
  <rect
    className="undefined"
    height="380"
    rx="15"
    ry="15"
    width="380"
    x="10"
    y="10"
  />
  <g>
    <image
      href="/images/star_white.svg"
      width="390"
      x="-15"
      y="65"
    />
    <image
      className=""
      height="220"
      href="/images/types/type-builder.png"
      width="220"
      x="90"
      y="132"
    />
  </g>
  <g
    id="type"
  >
    <rect
      height="75"
      rx="15"
      ry="15"
      width="360"
      x="20"
      y="20"
    />
    <text
      dominantBaseline="hanging"
      textAnchor="middle"
      x="200"
      y="33"
    >
      ITEM
    </text>
  </g>
  <g>
    <rect
      height="100"
      rx="50"
      ry="50"
      width="100"
      x="302"
      y="302"
    />
    <text
      dominantBaseline="hanging"
      fill="black"
      textAnchor="middle"
      x="351"
      y="331"
    >
      x2
    </text>
  </g>
  <g>
    <rect
      height="100"
      rx="50"
      ry="50"
      width="100"
      x="0"
      y="302"
    />
    <image
      height="70"
      href="/images/ranks/gold.png"
      width="70"
      x="15"
      y="315"
    />
  </g>
  <defs>
    <filter
      colorInterpolationFilters="sRGB"
      height="200%"
      id="drop-shadow"
      width="200%"
      x="-50%"
      y="-50%"
    >
      <feGaussianBlur
        id="blur"
        in="SourceAlpha"
        result="SA-o-blur"
        stdDeviation="5"
      />
      <feComponentTransfer
        in="SA-o-blur"
        result="SA-o-b-contIN"
      >
        <feFuncA
          id="contour"
          tableValues="0 1"
          type="table"
        />
      </feComponentTransfer>
      <feComposite
        in="SA-o-blur"
        in2="SA-o-b-contIN"
        operator="in"
        result="SA-o-b-cont"
      />
      <feComponentTransfer
        in="SA-o-b-cont"
        result="SA-o-b-c-sprd"
      >
        <feFuncA
          id="spread-ctrl"
          slope="200"
          type="linear"
        />
      </feComponentTransfer>
      <feColorMatrix
        id="recolor"
        in="SA-o-b-c-sprd"
        result="SA-o-b-c-s-recolor"
        type="matrix"
        values="0 0 0 0 1 0 0 0 0 1 0 0 0 0 1 0 0 0 1 0"
      />
      <feMerge>
        <feMergeNode
          in="SA-o-b-c-s-r-mix"
        />
        <feMergeNode
          in="SourceGraphic"
        />
      </feMerge>
    </filter>
  </defs>
</svg>
`;
